chainer/training/updater.py

```diff
@@ -260,6 +273,15 @@ def __init__(self, iterator, optimizer, converter=convert.concat_examples,
             if devices['main'] >= 0:
                 optimizer.target.to_gpu(devices['main'])
 
+        # Correct optimizer parameters for new minibatch size
+        optim = optimizer.__class__.__name__
+        if optim in ('Adam', 'AdaGrad', 'RMSprop'):
+            optimizer.eps /= len(devices)
```

Comment:
Yes, you are right. `eps` should be multiplied for those three update rules. I was wrong about the change for AdaDelta, which might be impossible to fix by multiplying and dividing hyperparameters; RMSpropGraves should also multiply, not divide, `eps` by `n_gpus ** 2`.